### PR TITLE
fix: handle assets in CSS extraction

### LINKED ISSUE #162: babel-preset/webpack-loader: Build time transforms do not handle assets properly
### Repro steps

- Open sandbox, https://stackblitz.com/edit/sveltejs-kit-template-default-urdlga?file=src%2Findex.js (download locally if it works better)
- `yarn`
- `webpack`

### The problem

- Check output in `dist/main.*.js`
  image
  - Runtime version (i.e. without transforms) uses proper file
  - Build time version (i.e. after transforms) references source file

The problem is in module evaluation and missing special handling around it to keep imports in place.

### PR DESCRIPTION
**Pull request by layershifter**
Fixes #162.

This PR fixes assets handling for CSS extraction. Compared to the previous PR it does reverse transforms:

```ts
import { __styles } from '@griffel/react';
import _asset from './blank.jpg';

export const useStyles = __styles(
  {
    root: {
      Bcmaq0h: 'fnwsaxv',
    },
  },
  {
    d: [`.fnwsaxv{background-image:url(${_asset});}`],
  },
);
// ⬇️⬇️⬇️
import { __styles, __css } from '@griffel/react';

export const useStyles = __css({
  root: {
    Bcmaq0h: 'fnwsaxv',
  },
  // Rules are remove by the plugin, but it's a result of added code: it inlines imports
  // {
  //   d: [`.fnwsaxv{background-image:url(${'../__fixtures__/webpack/assets/blank.jpg'});}`],
  // },
});
```

All work is done in `babelPluginStripGriffelRuntime`:
- removes imports for assets used in `__styles` calls
- inlines imports used in `__styles` calls

Imports are inlined with relative path to `griffel.css` file so `css-loader` could handle them properly and extract assets.

### CODE REVIEW

File: packages/webpack-extraction-plugin/src/GriffelCSSExtractionPlugin.test.ts [hunk 1]
```diff
@@ -34,6 +34,7 @@ async function compileSourceWithWebpack(
       path: path.resolve(__dirname),
       filename: 'bundle.js',
       pathinfo: false,
+      assetModuleFilename: '[name][ext]',
```
Comment (layershifter) [inline packages/webpack-extraction-plugin/src/GriffelCSSExtractionPlugin.test.ts:37]: Configure explicitly to have original filenames.

File: packages/webpack-extraction-plugin/src/babelPluginStripGriffelRuntime.test.ts [hunk 2]
```diff
@@ -16,9 +16,7 @@ pluginTester({
     },
   },
   pluginOptions: {
-    babelOptions: {
```
Comment (layershifter) [inline packages/webpack-extraction-plugin/src/babelPluginStripGriffelRuntime.test.ts:19]: Old options were copy-paste issue.

File: packages/webpack-extraction-plugin/src/babelPluginStripGriffelRuntime.test.ts [hunk 3]
```diff
@@ -16,9 +16,7 @@ pluginTester({
...
-      presets: ['@babel/typescript'],
-    },
+    resourceDirectory: process.cwd(),
```
Comment (layershifter) [inline packages/webpack-extraction-plugin/src/babelPluginStripGriffelRuntime.test.ts:19]: This value is just for testing, Webpack loader will pass the real one.

File: packages/webpack-extraction-plugin/src/babelPluginStripGriffelRuntime.ts [hunk 4]
```diff
@@ -79,6 +115,59 @@ export const babelPluginStripGriffelRuntime = declare<
...
+                      // When imports are inlined, we need to adjust the relative paths inside url(..) expressions
+                      // to allow css-loader resolve an imported asset properly
+                      transformUrl(
+                        state.filename!,
+                        options.resourceDirectory!,
+                        importDeclarationPath.get('source').node.value,
+                      ),
+                    ),
+                  );
+                  importDeclarationPath.remove();
```
Comment (ling1726) [inline packages/webpack-extraction-plugin/src/babelPluginStripGriffelRuntime.ts:167]: why remove the import decalartion ? wouldn't it also get stripped by bundlers ?

File: packages/webpack-extraction-plugin/src/GriffelCSSExtractionPlugin.test.ts (hunk 1 above)
Comment (ling1726) [reply packages/webpack-extraction-plugin/src/GriffelCSSExtractionPlugin.test.ts:37]: will end users need to have this in their webpack configs ?

**Review by ling1726 — APPROVED** [review]
A few small questions but otherwise it's quite clear that you're using babel to find the import path of the inline template literal asset and just doing a replacement

File: packages/webpack-extraction-plugin/src/GriffelCSSExtractionPlugin.test.ts (hunk 1 above)
Comment (layershifter) [reply packages/webpack-extraction-plugin/src/GriffelCSSExtractionPlugin.test.ts:37]: Nope, it's just to have the same names for assets in `fs.json` files uses in tests.

File: packages/webpack-extraction-plugin/src/babelPluginStripGriffelRuntime.ts (hunk 4 above)
Comment (layershifter) [reply packages/webpack-extraction-plugin/src/babelPluginStripGriffelRuntime.ts:167]: Technically these is no reason, but as these imports are used only for a specific declaration I decided to remove it to be a bit nicer 🐱

**Pull request merged**